src/shared/configUtils.ts

```diff
@@ -7,12 +7,38 @@
 
 import { Workspace } from '@lwc/lwc-dev-server';
 import { CryptoUtils, SSLCertificateData } from '@salesforce/lwc-dev-mobile-core';
-import { Config, ConfigAggregator } from '@salesforce/core';
+import { Config, ConfigAggregator, Connection } from '@salesforce/core';
 import configMeta, { ConfigVars, SerializedSSLCertificateData } from './../configMeta.js';
 
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export interface IdentityTokenService {
+  saveTokenToServer(token: string): Promise<string>;
+}
+
+export class AppServerIdentityTokenService implements IdentityTokenService {
```

Comment:
It seems that `AppServerIdentityTokenService` is only used in this file so it should be made private and not exported.

In fact, I think we should get rid of `IdentityTokenService` interface and instead pass a `Connection` object to `getOrCreateIdentityToken` and `getOrCreateIdentityToken` would then use a private helper function to accomplish its task of saving token to server.

Then in your tests, instead of passing in a `TestIdentityTokenService` you could pass in mock `Connection` and also stub out `connection.sobject` and `sobject.insert`

Or did you do it this way b/c you were hitting test issues?

Reply:
Yes, this was done intentionally. Remember the issue 
```
TypeError: Attempted to wrap request which is already wrapped 
```
This was caused by sinon not being able to stub `Connection` which @khawkins helped identify. After consulting with him this was the simple approach that accomplished what we need to do.

Reply:
Resolving as we talked about this on slack earlier.